spec/app.spec.js

```diff
@@ -79,6 +102,115 @@ describe('app.spec: app namespace tests', () => {
             `);
             assert.equal(runner.getOutput(), 'NE_RT_NATRTER');
         });
-    });
 
+        it('throws an error for an invalid event name', async () => {
+            runner.run(`
+                try {
+                    await Neutralino.app.broadcast(12345); 
+                } catch(err) {
+                    await __close(err.code);
+                }
+        `   );
+            assert.equal(runner.getOutput(), 'NE_RT_NATRTER'); 
+        });
+
+        it('broadcasts an event with no listeners registered', async () => {
+            runner.run(`
+                try {
+                    await Neutralino.app.broadcast('eventWithNoListener');
+                    await __close('success');
+                }
+                catch(err) {
+                    await __close('failed');
+                }
+        `   );
+            assert.equal(runner.getOutput(), 'success');
+        });
+
+        it('handles high load by broadcasting multiple events rapidly', async () => {
+            runner.run(`
+                try {
+                    for (let i = 0; i < 1000; i++) {
+                        Neutralino.app.broadcast('testEvent' + i);
+                    }
+                    await __close('success');
+                } catch (err) {
+                await __close('failed');
+            }
+            `);
+            assert.equal(runner.getOutput(), 'success');
+        });
+
+        it('triggers all registered listeners when broadcasting an event', async () => {
+            runner.run(`
+                let listener1Triggered = false;
+                let listener2Triggered = false;
+            
+                function listener1() {
+                    listener1Triggered = true;
+                }
+            
+                function listener2() {
+                    listener2Triggered = true;
+                }
+            
+                await Neutralino.events.on('multiListenerEvent', listener1);
+                await Neutralino.events.on('multiListenerEvent', listener2);
+            
+                await Neutralino.app.broadcast('multiListenerEvent');
```

Comment:
Are you sure this works always? Seems like `app.broadcast()` can end before the app trigger all the above listeners by keeping `false` for both variables. Do we need to add `__close()` within these listeners? Please research a bit more about this scenario

Reply:
Hey, since the listeners will be triggered in order I think adding `__close()` in the second listener should be a viable solution.